kamon-spray/src/main/scala/kamon/spray/instrumentation/ClientRequestInstrumentation.scala

```diff
@@ -123,9 +123,7 @@ class ClientRequestInstrumentation {
 
         request.asInstanceOf[SegmentAware].segment = segment
 
-        val responseFuture = originalSendReceive.apply(request)
-        responseFuture.map(result ⇒ segment.finish())(SameThreadExecutionContext)
-        responseFuture
+        originalSendReceive.apply(request).andThen { case _ ⇒ segment.finish() }(SameThreadExecutionContext)
```

Comment:
I saw your commit on play WS stuff 58919a59acd867bc919cbbb24d13613f666f9add and for the sake of code consistency I will update the PR, although I think andThen is better because we make sure we finish the segment as soon the request completes